tests/integration/test_integration.py

```diff
@@ -0,0 +1,110 @@
+        test_name = self.id().split('.')[-1]    # `id()` returns something like `test_integration.TestIntegration.test_set1`
+        data_dir = fixtures_dir / test_name
+        shutil.copytree(data_dir / 'input', self.input_dir)
+        reference_source_dir = data_dir / 'reference'
+        if reference_source_dir.exists():
+            shutil.copytree(data_dir / 'reference', self.reference_dir)
+
+    def test_set1(self):
+        helper.main(['-i', str(self.input_dir), '-o', str(self.output_dir)])
+        for file in self.output_dir.rglob('*'):
+            reference = self.reference_dir / file.name
+            self.assertFileMatches(file, reference)
+
+    def test_current_reference_collection(self):
+        helper.main(['-i', str(self.input_dir), '-o', str(self.output_dir), '--divide-to-dates'])
+        for file in self.output_dir.rglob('*'):
+            if not file.is_file():
+                continue
+            reference = self.reference_dir / file.relative_to(self.output_dir)
+            self.assertFileMatches(file, reference)
+
+    def assertFileMatches(self, file, reference):
+        self.assertFileDates(file, reference)
+        self.assertExifData(file, reference)
+
+    def assertFileDates(self, file, reference):
+        stats = os.stat(file)
+        reference_stats = os.stat(reference)
+        self.assertEqual(stats.st_birthtime, reference_stats.st_birthtime, f'File creation date not matching for {file}')
+        self.assertEqual(stats.st_mtime, reference_stats.st_mtime, f'File modification date not matching for {file}')
+
+    def assertExifData(self, file, reference):
+        file_exif = None
+        reference_exif = None
+        try:
+            file_exif = piexif.load(str(file))
+        except Exception as e:
+            pass
+        try:
+            reference_exif = piexif.load(str(reference))
+        except Exception as e:
+            pass
+
+        if file_exif is None and reference_exif is None:
+            # it's okay if both files don't have EXIF data
+            return
+        self.assertEqual(file_exif is None, reference_exif is None, f'EXIF presence mismatch for {file}')
+
+        # show long diffs
+        self.maxDiff = None
+
+        # remove the thumbnails to keep the diff smaller
+        del file_exif['thumbnail']
+        del reference_exif['thumbnail']
+       
+        # see https://www.exiv2.org/tags.html for tag details
+        TAG_DATE_TIME = 306
+        tags = [
+            ['0th', piexif.ImageIFD.DateTime], 
+            ['Exif', piexif.ExifIFD.DateTimeOriginal], 
+            ['Exif', piexif.ExifIFD.DateTimeDigitized],
+            ['GPS', piexif.GPSIFD.GPSVersionID],
+            ['GPS', piexif.GPSIFD.GPSLatitude],
+            ['GPS', piexif.GPSIFD.GPSLatitudeRef],
+            ['GPS', piexif.GPSIFD.GPSLongitude],
+            ['GPS', piexif.GPSIFD.GPSLongitudeRef],
+            ['GPS', piexif.GPSIFD.GPSAltitude],
+            ['GPS', piexif.GPSIFD.GPSAltitudeRef]
```

Comment:
I see you added a lot of `piexif` stuff - I'm worried about this, because I wanted to start transitioning to other library - because this one is pretty dead - and now I have more stuff to make me procrastinate this decision, and more stuff that can break :(

Reply:
Do you already have one in mind? If you haven't decided yet, I don't think that should block the merge of this PR. On the contrary, it makes sense to have a testing rig in place when swapping out integral parts to be sure everything keeps working. Also the unit tests aren't coupled to the implementation: we can replace the EXIF lib in the implementation and keep the tests on `piexif` for a while and see how it goes. Or the other way round.

Also I contain all the `piexif` stuff in this one method - I can also move the import down here if you'd like that to be more explicit.

Reply:
[exif](https://pypi.org/project/exif/) looks nicer and actually maintained

Reply:
How about the following plan:

1. Get the test setup in place
2. Add a couple of more tests to increase the confidence
3. Switch the tests over to `exif` - in a separate PR so the change is isolated and it can easily be judged if the switch broke anything
4. After gaining some experience with `exif` switch over the `implementation` or try another lib in the tests

?